src/main/java/rx/internal/operators/OperatorPublish.java

```diff
@@ -15,367 +15,736 @@
+                        }
+                        
+                        // it may happen everyone has unsubscribed between here and producers.get()
+                        // or we have no subscribers at all to begin with
+                        if (len == unsubscribed) {
+                            // so let's consume a value from the queue
+                            Object v = queue.poll();
+                            // or terminate if there was a terminal event and the queue is empty
+                            if (checkTerminated(terminalEvent, v == null)) {
+                                skipFinal = true;
+                                return;
+                            }
+                            // otherwise, just ask for a new value
+                            request(1);
+                            // and retry emitting to potential new child-subscribers
+                            continue;
+                        }
+                        // if we get here, it means there are non-unsubscribed child-subscribers
+                        // and we count the number of emitted values because the queue
+                        // may contain less than requested
+                        int d = 0;
+                        while (d < maxRequested) {
+                            Object v = queue.poll();
+                            empty = v == null;
+                            // let's check if there is a terminal event and the queue became empty just now
+                            if (checkTerminated(terminalEvent, empty)) {
+                                skipFinal = true;
+                                return;
+                            }
+                            // the queue is empty but we aren't terminated yet, finish this emission loop
+                            if (empty) {
+                                break;
+                            }
+                            // we need to unwrap potential nulls
+                            T value = nl.getValue(v);
+                            // let's emit this value to all child subscribers
+                            for (InnerProducer<T> ip : ps) {
+                                // if ip.get() is negative, the child has either unsubscribed in the
+                                // meantime or hasn't requested anything yet
+                                // this eager behavior will skip unsubscribed children in case
+                                // multiple values are available in the queue
+                                if (ip.get() > 0L) {
+                                    try {
+                                        ip.child.onNext(value);
+                                    } catch (Throwable t) {
+                                        // we bounce back exceptions and kick out the child subscriber
+                                        ip.unsubscribe();
+                                        ip.child.onError(t);
```

Comment:
This is what happens, yes. I'm not sure if we want to go that deep and paranoid about onError and onCompleted.